src/SDK/Trace/Behavior/SpanExporterTrait.php

```diff
@@ -44,7 +54,65 @@ public function export(iterable $spans): int
             return SpanExporterInterface::STATUS_SUCCESS;
         }
 
-        return $this->doExport($spans); /** @phpstan-ignore-line */
+        // If retryPolicy is not set then just doExport once and return the status
+        if ($this->retryPolicy === null) {
+            return $this->doExport($spans);
+        }
+
+        while ($shouldRetry) {
+            // find the delay time before retry (it is 0 for the first attempt)
+            $delay = $this->retryPolicy->getDelay($attempt++);
+            if ($attempt > 0) {
+                self::logDebug('Waiting for ' . $delay . ' seconds before retrying export');
+                sleep($delay);
```

Comment:
I don't think sleeping here is a good idea, because it will cause the application being instrumented to wait. A better way might be to keep track of spans to retry (eg in internal storage), and check for spans to retry as a side-effect of doing something else in the exporter. We could do a final check as part of exporter shutdown to see if any pending spans need to be exported